packages/react-instantsearch-hooks/src/hooks/useConnector.ts

```diff
@@ -89,17 +89,19 @@ export function useConnector<
     if (widget.getWidgetRenderState) {
       // The helper exists because we've started InstantSearch.
       const helper = parentIndex.getHelper()!;
+      const uiState = parentIndex.getWidgetUiState({})[
+        parentIndex.getIndexId()
+      ];
+      helper.state =
+        widget.getWidgetSearchParameters?.(helper.state, { uiState }) ||
+        helper.state;
```

Comment:
this means that if this part of the function gets called, but the widget never actually mounts, there will be stray search parameters not associated to a widget (as we don't call dispose). Could this ever be a real issue?

Reply:
Why wouldn't the widget mount at this point?

And `getWidgetSearchParameters()` returns a value without triggering side effects, so I'm not sure to understand what you're saying.

Reply:
getWidgetSearchParameters doesn't have any side-effects, but `helper.state = getWidgetSearchParameters()` is a side-effect. 

I'm not sure why it wouldn't be mounted, but I'm not 100% sure whether the contract of react allows a memo to be called before/after the component is already mounted? Are there any specs of that? Is it possible a memo is called without the rest of the component being called?

Reply:
Assigning `helper.state` is indeed a side effect. I think it's localized in that case because each widget overrides it for its initial rendering, and then the InstantSearch.js lifecycle takes over and sets it.

React can call memo anytime that it wants to call it (for memory/optimization purposes), so there's no certainty that it'll be called only once. I don't think that the new code introduced competes with the existing logic of `useConnector()` and `useMemo()` though. We already had this consideration before this change.